src/loaders/httpStore.js

```diff
@@ -0,0 +1,41 @@
+import { KeyError, HTTPError } from 'zarr';
+
+/**
+ * Preserves (double) slashes earlier in the path, so this works better
+ * for URLs. From https://stackoverflow.com/a/46427607/4178400
+ * @param args parts of a path or URL to join.
+ */
+export function joinUrlParts(...args) {
+  return args.map((part, i) => {
+    if (i === 0) {
+      return part.trim().replace(/[\/]*$/g, '');
+    } else {
+      return part.trim().replace(/(^[\/]*|[\/]*$)/g, '');
+    }
+  }).filter(x=>x.length).join('/');
+}
+
+export default class HTTPStore {
+  constructor(url, options = {}) {
+    this.url = url;
+    this.options = options;
+  }
+
+  async getItem(key, options = {}) {
+    const url = joinUrlParts(this.url, key);
+    const value = await fetch(url, { ...this.options, ...options });
```

Comment:
It might be worth proposing adding this the actual zarr.js HTTPStore implementation. It's just one extract argument and allows this more granular control over the `store.getItem` that's a much lower-level function.

https://github.com/gzuidhof/zarr.js/pull/58

Reply:
Ok, but this shouldn't block us?

Reply:
Nope. I'll mention this there